CHANGELOG.md

```diff
@@ -1,13 +1,166 @@
+
+```sh
+# Update helm
+curl https://raw.githubusercontent.com/kubernetes/helm/master/scripts/get | bash
+
+# Update tiller (on the cluster)
+helm init --upgrade --service-account=tiller
+
+# Verify the update
+# NOTE: you may need to cancel and re-run the command, it should work within 30
+#       seconds.
+helm version
+# VERIFY: Did it return both the client and server version?
+# Client: &version.Version{SemVer:"v2.10.0", GitCommit:"9ad53aac42165a5fadc6c87be0dea6b115f93090", GitTreeState:"clean"}
+# Server: &version.Version{SemVer:"v2.10.0", GitCommit:"9ad53aac42165a5fadc6c87be0dea6b115f93090", GitTreeState:"clean"}
+```
+
+#### 3. Clean up pre-puller resources
+
+```sh
+# The script will delete resources that were meant to be temporary
+# The bug that caused this is fixed in version 0.7.0-beta.1 of the Helm chart
+NAMESPACE=<YOUR-NAMESPACE>
+
+resource_types="daemonset,serviceaccount,clusterrole,clusterrolebinding,job"
+for bad_resource in $(kubectl get $resource_types --namespace $NAMESPACE | grep '/pre-pull' | awk '{print $1}');
+do
+    kubectl delete $bad_resource --namespace $NAMESPACE --now
+done
+
+kubectl delete $resource_types --selector hub.jupyter.org/deletable=true --namespace $NAMESPACE --now
+```
+
+#### 4. Clean up problematic revisions in your Helm release
+
+```sh
+# Look up the name of your Helm release (installation of a Helm chart)
+helm list
+
+# Store the name of the Helm release
+RELEASE_NAME=<YOUR-RELEASE-NAME>
+
+# Give yourself an overview of this release's revisions
+helm history $RELEASE_NAME
+
+# Check if you have multiple revisions in a DEPLOYED status (a bug), or if you
+# have old PENDING_UPGRADES or FAILED revisions (may be problematic).
+helm history $RELEASE_NAME | grep --extended-regexp "DEPLOYED|FAILED|PENDING_UPGRADE"
+
+# If you have multiple revisions in DEPLOYED status, you this script will clean
+# up all configmaps except the latest with DEPLOYED status.
+deployed_revisions=($(helm history $RELEASE_NAME | grep DEPLOYED | awk '{print $1}'))
+for revision in ${deployed_revisions[@]::${#deployed_revisions[@]}-1};
+do
+    kubectl delete configmap $RELEASE_NAME.v$revision --namespace kube-system
+done
+
+# It seems plausible that upgrade failures could have to do with revisions
+# having a PENDING_UPGRADE or FAILED status in the revision history. To delete
+# them run the following command.
+kubectl delete configmap --selector "NAME=$RELEASE_NAME,STATUS in (FAILED,PENDING_UPGRADE)" --namespace kube-system
+```
+
+#### 5. Make the upgrade
+```sh
+RELEASE_NAME=<YOUR-RELEASE-NAME>
+NAMESPACE=<YOUR-NAMESPACE>
+
+helm repo add jupyterhub https://jupyterhub.github.io/helm-chart/
+helm repo update
+
+# NOTE: We need the --force flag to allow recreation of resources that can't be
+#       upgraded to the new state by a patch.
+helm upgrade $RELEASE_NAME jupyterhub/jupyterhub --install \
+    --force \
+    --version=0.7.0-beta.1 \
+    --namespace=$NAMESPACE \
+    --values config-dev.yaml \
```

Comment:
We use `config.yaml` in most of the rest of the docs, use that here as well?

Reply:
Ooops absolutely